src/main.rs

```
@@ -59,6 +70,112 @@ fn check_valid_env(
+        let command = command_args.value_of("COMMAND").unwrap();
+        sub_proc = Exec::shell(command);
+    } else if let Some(arg_args) = subcmd_args.subcommand_matches("arguments") {
+        let mut arguments = arg_args.values_of_lossy("ARGUMENTS").unwrap();
+        let command = arguments.remove(0);
+        if command.contains(' ') {
+            warn("command contains spaces, and will likely fail.".to_string());
+            let mut reformed = format!("{} {}", command, arguments.join(" "));
+            reformed = reformed.replace("$", "\\$");
+            println!("Try using 'cloudtruth run command \"{}\"'", reformed.trim());
+        }
+        sub_proc = Exec::cmd(command).args(&arguments);
+    } else {
+        warn_missing_subcommand("run");
+        process::exit(0);
+    }
+
+    // setup the environment for the sub-process
+    let preserve = subcmd_args.is_present("preserve");
+    let overrides = subcmd_args.values_of_lossy("set").unwrap_or_default();
+    let removals = subcmd_args.values_of_lossy("remove").unwrap_or_default();
+    let mut env_vars = if !preserve {
+        HashMap::new()
+    } else {
+        current_env()
+    };
+
+    // add breadcrumbs about which environment this came from
+    env_vars.insert("CT_ENV".to_string(), env.unwrap_or("default").to_string());
+
+    // merge in the items from the CloudTruth environment
+    env_vars.extend(get_ct_vars(org_id, env, environments)?);
+
+    // add the overrides
+    env_vars.extend(process_overrides(overrides));
+
+    // remove the specified values
+    for r in removals {
+        env_vars.remove(r.as_str());
+    }
+
+    // Common setup for the subprocess. By default, it streams stdin/stdout/stderr to parent.
+    sub_proc = sub_proc.env_clear();
+    for (key, value) in env_vars {
+        sub_proc = sub_proc.env(key, value);
+    }
+
+    // now, run it and wait for the result
+    let status = sub_proc.join();
+    println!("Exited with status {:?}", status);
```

Is this intentional? `println!` statements are swallowed in tests, unless there's a test failure. So, adding them to tests is fine.

---

I removed the statement... Although it is important to me, I suspect our users will just find it annoying.